Use the default template's background color for swatches

joelostblom opened the pull request:
This is a suggestion to use the default template's background color for swatches Instead of always using the plotly template. The potential benefit of this change would be that it is easier to tell which colors sequence go well with the currently used template.

It would be neat if this also checked if there was a PX template set, and I asked a related question about how PX and PIO templates synergize in #1869.

Screenshots:

![image](https://user-images.githubusercontent.com/4560057/68197772-1226a300-ffbb-11e9-9042-b7b7ab772807.png)
---

![image](https://user-images.githubusercontent.com/4560057/68198161-ce806900-ffbb-11e9-84c6-0860cda6c254.png)
---

![image](https://user-images.githubusercontent.com/4560057/68198029-924d0880-ffbb-11e9-8a89-352020257189.png)
---

![image](https://user-images.githubusercontent.com/4560057/68198089-ab55b980-ffbb-11e9-8d30-09317b7c70cb.png)
---

![image](https://user-images.githubusercontent.com/4560057/68198134-bdcff300-ffbb-11e9-975d-657dc9db3065.png)
---

![image](https://user-images.githubusercontent.com/4560057/68197994-85c8b000-ffbb-11e9-8391-a3b10b04fd92.png)

nicolaskruchten commented:
Looks good to me if you can fix the formatting problem in CI?

joelostblom commented:
I believe this is the same CI error as on the other PRs and unrelated to the changes I have made. The error is

```
error: cannot format /home/circleci/project/packages/python/plotly/plotly/basedatatypes.py: cannot use --safe with this file; failed to parse source file.  AST error message: misplaced type annotation (<unknown>, line 256)
```
I think that [line 256](https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/basedatatypes.py#L256) need to have `collections.` added in front of `OrderedDict`

nicolaskruchten commented:
If you want to have this check the `px.defaults.template` and use that I'd be fine with it.

joelostblom commented:
Thanks, added!

I also added the default width, so that the autoadjusting behavior can be overridden if desired.

width 600 samples:
![image](https://user-images.githubusercontent.com/4560057/68270571-e9a3b500-005d-11ea-8158-92fb716acceb.png)


![image](https://user-images.githubusercontent.com/4560057/68270426-7c901f80-005d-11ea-848f-80c006dfeb8c.png)

nicolaskruchten commented:
re the synergization of px and pio templates... Really the way to do this would be to rewrite this little swatch function on top of `px.bar` somehow :)

nicolaskruchten commented:
the plot titles are also pretty ugly at the moment... they used to say `px.colors.whatever` but after to move into plotly.py, things got shuffled around

joelostblom commented:
Rebased and cleaned up the titles 

![image](https://user-images.githubusercontent.com/4560057/68320980-2e146c80-00c1-11ea-9968-1e221eab4612.png)

emmanuelle commented:
Closing since #1886 was merged, thanks @joelostblom !
